lib/btc-utils.js

```diff
@@ -136,7 +135,7 @@ const waitForBitcoinTxToBeInMempool = async (btcTxHelper, btcTxHash, maxAttempts
       throw e;
     };
   
-    const { result: btcTxAlreadyFoundInMempool } = retryWithCheck(
+    const { result: btcTxAlreadyFoundInMempool } = await retryWithCheck(
```

Comment:
Bug 2 in this `waitForBitcoinTxToBeInMempool` function. It didn't have an `await`.